### DISCUSSION TITLE
When image_threshold and pixel_threshold are generated?

### DISCUSSION THREAD

**Discussion by glucasol** [open-edge-platform/anomalib#832] [Q&A]
Hi all,

I have a doubt about the threshold values.
I think the values of image_threshold and pixel_threshold that are used to calculate the predicted label and predicted mask are calculated at the validation part ( witch is part of the training) and these values are used to make the predictions on test part.
If this is true, the threshold value requires normal and abnormal images to be set.

Am I wright? Please, correct me if I am wrong.

Thanks.

**Comment by samet-akcay** [comment]
Hi @glucasol, yes, you are right.  In order to properly set the thresholds, anomalib needs anomalous images. The more abnormal images, the threshold gets better. If there is no abnormal image provided to the pipeline, anomalib used to complain about this and didn't finish the training.

We have recently merged PR #822, which adds synthetic anomaly generation for validation and evaluation purposes. So, if there is no abnormal images in your dataset, anomalib could generate some synthetic anomalous images to compute a threshold. Since this has recently been merged, there is unfortunately no documentation for this. 

I'll shortly add more instructions regarding how this could be done.

**Comment by arkerman** [comment]
Hi, @samet-akcay ! I have some extra questions about the pixel_threshold. 
When i trained reverse_distillation on dataset MVTec, I export torch model(model.pt). Then i use ```torch.load()``` to load this .pt file. I found there are two keys in it. The keys are as follows : 'model' and 'metadata'. The value of ’metadata‘ contains piexl_threshold, image_threshold, original image size, etc. 
So, my question is that how did pixel_threshold calculate in reverse_distillation? And how did it be export to the .pt file? It seems that the pixel_threshold is calculated by dynamicly and adaptively. And i can't find the source code about this. 
Any suggestions will be appreaciated.
